source/isaaclab/isaaclab/managers/observation_manager.py

```diff
@@ -330,7 +336,13 @@ def compute_group(self, group_name: str) -> torch.Tensor | dict[str, torch.Tenso
 
         # concatenate all observations in the group together
         if self._group_obs_concatenate[group_name]:
-            return torch.cat(list(group_obs.values()), dim=-1)
+            # set the concatenate dimension, account for the batch dimension if positive dimension is given
+            dim = (
+                self._group_obs_concatenate_dim[group_name] + 1
+                if self._group_obs_concatenate_dim[group_name] >= 0
+                else self._group_obs_concatenate_dim[group_name]
+            )
```

Comment:
This if-else is not needed (or can just be done once when the config is parsed). Would keep the code simple here:

```python
return torch.cat(list(group_obs.values()), dim=self._group_obs_concatenate_dim[group_name])
```